src/components/QuestionnaireForm/QuestionnaireForm.jsx

```diff
@@ -1051,6 +1053,39 @@ export default class QuestionnaireForm extends Component {
     }
   }
 
+  updateSavedResponseWithPrepopulation = (newOne, saved) => {
+    newOne.item.map(newItem => {
+      let savedIndex = saved.item.findIndex(savedItem => newItem.linkId == savedItem.linkId);
+      if (savedIndex != -1) {
+        updateItem(newItem, saved.item[savedIndex]);
+      }
+    })
+
+    function updateItem(newItem, savedItem) {
+      if (newItem.item == undefined) {
+        //find the corresponding linkId in savedItem and replace it
+        function replaceItem(savedItem) {
+          if (savedItem.item == undefined) {
+            // Couldn't found linkId in the savedItem, ignore it
+            return;
+          }
+          let foundSavedItemIndex = savedItem.item.findIndex(saved => saved.linkId == newItem.linkId);
+          if (foundSavedItemIndex != -1) {
+            savedItem.item[foundSavedItemIndex] = newItem;
+            return;
+          } else {
+            replaceItem(savedItem.item);
+          }
+        };
+        replaceItem(savedItem);
+      } else {
+        newItem.item.forEach(newSubItem => {
+          updateItem(newSubItem, savedItem);
```

Comment:
I think the problem may be due to the fact that the `newItem` is being searched recursively, but the `savedItem` is not.

Reply:
The issues should be fixed now. Good catch! @KeeyanGhoreshi